### PR TITLE
[Test Only] Support ConvTranspose-14

### PR DESCRIPTION
**Pull request by jiafatom**
**Description**: Support ConvTranspose-14.

**Motivation and Context**
This change aligns the keras/tensorflow ConvTranspose padding=same case for dynamic input shape. It handles stride>1 in particular. Validated two models locally (unetplusplus and unet) and it works correctly.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h [hunk 1]
```diff
@@ -212,7 +212,7 @@ struct ConvTransposeAttributes : public ConvAttributes {
     if (pad_type == AutoPadType::SAME_UPPER || pad_type == AutoPadType::SAME_LOWER) {
       // total pad
       auto total_pad = ComputeTotalPad(in_size, stride, adj,
-                                       kernel, dilation, in_size);
+                                       kernel, dilation, in_size * stride);
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h:215]: Do all 4 models needing this give results aligning with tf after this ?

Comment (jiafatom) [reply onnxruntime/core/providers/cpu/nn/conv_transpose_attributes.h:215]: Two models are working good with this change. The other two models have other issues which are unrelated.

**Comment by hariharans29** [comment]
Closing as https://github.com/microsoft/onnxruntime/pull/6305 has tests needed for this and will be checked-in
